system/jhtree/jhtree.cpp

```diff
@@ -229,13 +229,17 @@ bool SegMonitorList::incrementKey(unsigned segno, void *keyBuffer) const
     }
 }
 
-unsigned SegMonitorList::_lastRealSeg() const
+unsigned SegMonitorList::_lastRealSeg()
 {
+    unfiltered = false;
```

Comment:
I hesitated about this change - setting unfiltered as a side-effect.  I think it is ok.  Probably clearer if this was inlined into the only place that calls it.  But fas as is.

Reply:
The only place that calls it doesn't know - can't tell difference between 0 filters and 1...